src/components/App.js

```diff
@@ -24,21 +25,26 @@ class App extends Component {
     const lat =  Number(user.data.attributes.lat)
     const lng = Number(user.data.attributes.long)
     const truckList = this.createLocationList(user)
-    this.setState({lat: lat, lng: lng, truckList:[...this.state.truckList, ...truckList ], trucks:[...this.state.trucks, ...user.data.attributes.trucks]})
+    this.setState({ lat: lat, lng: lng, truckList:[...this.state.truckList, ...truckList ] })
   }
 
   loginUser = (userName) => {
     fetchUserName(userName)
     .then(data => {
-      console.log('id', data.data.id)
+      console.log('user', data.data)
       const id = data.data.id
       fetchTrucks(id)
       .then(trucks => {
         console.log('trucks', trucks.data)
-        this.setState({ trucks: trucks.data})
+        this.sortByDistance(trucks.data)
       })
     })
-    .catch(error => console.log(error))
+    .catch(error => this.setState({ error: error.message }))
+  }
+
+  sortByDistance = (trucks) => {
+    const sortedTrucks = trucks.sort((a, b) => a.attributes.distance - b.attributes.distance)
+    this.setState({ trucks: sortedTrucks })
```

Comment:
could move this into a utility file in the future when all the filter methods are flushed out, we can rename this "filter trucks" and then use the sortByDistance method when distance is clicked

Reply:
How would that work with setting state?

Reply:
we would still set state within the filter function, and use the return of the helper functions